services/security/authentication/casigned.go

```
@@ -70,57 +69,63 @@ func (c *client) Login(ctx context.Context, group string, identity *security.Ide
 		time.Sleep(sleepTime)
-		log.Printf("Attempting to renew certificate\n")
+		logger.Info("Attempting to renew certificate")
 		err = auth.RenewCertificates(server, auth.GetWssdConfigLocation())
 		if err != nil {
 			// If certificate is expired, we attempt to re-login with set login config
 			if errors.IsExpired(err) {
-				fmt.Fprintf(os.Stderr, "Certificate Expired, Attemptin re-login %v", err)
+				logger.Error(err, "Certificate expired, attempting re-login")
 				err = reLoginOnExpiry(ctx, loginConfig, group, server)
 				if err == nil {
-					log.Println("Re-Login successful")
+					logger.Info("Re-login successful")
 					renewalAttempt = 0
 					continue
 				} else {
-					fmt.Fprintf(os.Stderr, "Re-Login Failure %v", err)
+					logger.Error(err, "Re-login failure")
 				}
 			}
-			renewalAttempt += 1
-			log.Printf("Failed to renew cert: %v. Attempts %d\n", err, renewalAttempt)
-			log.Printf("Certificate Expiry %s, Now %s\n", expiry.UTC().String(), time.Now().UTC().String())
+			renewalAttempt++
+			logger.Error(err, "Failed to renew certificate", "attempts", renewalAttempt)
+			logger.Info("Certificate expiry details", "expiry", expiry.UTC().String(), "now", time.Now().UTC().String())
 			time.Sleep(renewalBackoff)
 			continue
 		}
-		//reset renewalAttempt after successful renewal
+		// Reset renewalAttempt after successful renewal
 		renewalAttempt = 0
-		log.Println("Certificate renewal complete")
+		logger.Info("Certificate renewal complete")
 	}
 }
 
 // Get methods invokes the client Get method
 func (c *client) LoginWithConfig(ctx context.Context, group string, loginconfig auth.LoginConfig, enableRenewRoutine bool) (*auth.WssdConfig, error) {
+	logger := log.FromContext(ctx) // Retrieve the logger from context
+	if logger.GetSink() == nil {   // Check if the logger has a valid sink
+		logger = log.Log // Use the global default logger as a fallback
+	}
 
+	logger.Info("Generating client CSR")
```

These are going to be printed in moccti. We should limit the logger to the go routine. This function should return the appropriate errors rather than logging

---

done!